interop-testing/src/test/java/io/grpc/ChannelAndServerBuilderTest.java

```diff
@@ -48,19 +49,39 @@
    */
   @Parameters(name = "class={0}")
   public static Collection<Object[]> params() throws Exception {
+    @SuppressWarnings("unchecked")
```

Comment:
This somewhat defeats the purpose of this test.   It showed that every class that gRPC knows about overrides the static ctors, but since you enumerate it, new builders will be missed.    (Like the recently added Alts Channel)

Reply:
Not so. We have a hard-coded list, but it is _verified_ when possible (JDK < 9). So that means the list will be kept up-to-date. It is true that if we only ran on JDK 9 the list would likely get out-of-date, but in that case it seems like the best we could do is use the hard-coded list or delete the test. Looking at ClassPath.Scanner.getClassLoaderUrls, it seems impossible to fix for Java 9.

Reply:
If this doesn't work for Java 9, it might as well not run.    The list will get out of date, and it will give false confidence.

It would be better to check the java version and call Assume.assumeTrue(javaVersion() <= 8) in a BeforeClass.  If that isn't possible you could make a dummy test that asserts there is more than one class to check and throw an assumption exception otherwise.  

If this test doesn't work on java 9 then so be it.  That's okay.

Reply:
`@BeforeClass` runs after `params()`, so an assume in it doesn't work. Similarly, a failing assume in `params()` causes a test failure. And a dummy test doesn't work because no tests are run if `params()` returns an empty list.

I'm detecting when `getTopLevelClassesRecursive()` returns nothing and returning no params in that case. Note that returning an empty list of params causes the test class to be excluded from test reports, since by definition there were no tests within the class. So there isn't any "ignored" count or similar.